stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -1154,6 +1154,50 @@ swift_conformsToProtocolMaybeInstantiateSuperclasses(
   return {foundWitness, hasUninstantiatedSuperclass};
 }
 
+/// Determine if
+static bool isExecutingInIsolationOfConformance(
+    const Metadata *const type,
+    const ProtocolConformanceDescriptor *description,
+    const WitnessTable *table
+) {
+  // Resolve the global actor type.
+  SubstGenericParametersFromMetadata substitutions(type);
+  auto result = swift_getTypeByMangledName(
+     MetadataState::Abstract, description->getGlobalActorType(),
+     /*FIXME:conditionalArgs.data()*/{ },
+     [&substitutions](unsigned depth, unsigned index) {
+        return substitutions.getMetadata(depth, index).Ptr;
+      },
+    [&substitutions](const Metadata *type, unsigned index) {
+      return substitutions.getWitnessTable(type, index);
+    });
+  if (result.isError())
+    return false;
+
+  const Metadata *globalActorType = result.getType().getMetadata();
+  if (!globalActorType)
+    return false;
+
+  auto globalActorConformance = description->getGlobalActorConformance();
+  if (!globalActorConformance)
+    return false;
+
+  auto globalActorWitnessTable =
+      globalActorConformance->getWitnessTable(globalActorType);
+  if (!globalActorWitnessTable)
+    return false;
+
+  // The concurrency library provides a function to check whether we
+  // are executing on the given global actor.
+  auto isCurrentGlobalActor = SWIFT_LAZY_CONSTANT(reinterpret_cast<bool (*)(const Metadata *, const WitnessTable *)>(
+      dlsym(RTLD_DEFAULT, "swift_task_isCurrentGlobalActor")));
```

Comment:
Could we avoid this by having the concurrency library register a function with libswiftCore? `dlsym` is problematic because it won't work on Linux when we're statically linked (and it doesn't exist on Windows either).

Reply:
We already do this sort of thing for Concurrency types with standard mangling. Look at the uses of `ConcurrencyStandardTypeDescriptors`. We could extend that facility to a more general "register stuff from the Concurrency runtime" thing, or add a second one.

Reply:
That's a great idea. This is runtime SPI, so I'll generalize it rather than adding a second one. Fewer `__attribute__((constructor))`s is a good thing